cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java

```diff
@@ -141,6 +142,12 @@ public CassandraScheduler(
     public void registered(SchedulerDriver driver,
                            Protos.FrameworkID frameworkId,
                            Protos.MasterInfo masterInfo) {
+        // Return early if registered() has already been called once in the lifetime of this object.
+        if (!isSchedulerRegistered.compareAndSet(false, true)) {
```

Comment:
@verma7 Just curious here. As per point 3 in this:
```
1) Framework connects w/ master very first time --> registered()
2) Framework reconnects w/ same master after a zk blip --> reregistered()
3) Framework reconnects w/ failed over master --> registered()
4) Failed over framework connects w/ same master --> registered()
5) Failed over framework connects w/ failed over master --> registered()
```

Shouldn't the reconciler triggered again ? If yes, we can prevent re-init of objects but can still perform reconciliation.